decoder/src/frame.rs

```diff
@@ -269,13 +269,24 @@ fn format_args_real(
                                     format_u128(bitfields as u128, hint, &mut buf)?;
                                 }
                             }
-                            _ => format_u128(*x as u128, hint, &mut buf)?,
+                            _ => {
+                                if let Some(DisplayHint::Debug) = hint {
+                                    format_u128(*x as u128, parent_hint, &mut buf)?;
+                                } else {
+                                    format_u128(*x as u128, hint, &mut buf)?;
+                                }
+                            }
                         }
                     }
                     Arg::Ixx(x) => format_i128(*x as i128, hint, &mut buf)?,
                     Arg::Str(x) | Arg::Preformatted(x) => format_str(x, hint, &mut buf)?,
                     Arg::IStr(x) => format_str(x, hint, &mut buf)?,
-                    Arg::Format { format, args } => buf.push_str(&format_args(format, args, hint)),
+                    Arg::Format { format, args } => match parent_hint {
+                        Some(DisplayHint::Ascii) => {
```

Comment:
I don't quite follow why `:a` is a special case here. (there some bugs / limitations (?) around `:a` and generics but this doesn't seem like it fixes them or creates more)